src/csvcubed/writers/qbwriter.py

```diff
@@ -820,6 +982,45 @@ def _define_csvw_column_for_qb_column(
             column.structural_definition.__class__.__name__,
         )
 
+        obs_val_col: Optional[QbColumn[QbObservationValue]] = None
+        obs_val_cols = get_columns_of_dsd_type(self.cube, QbObservationValue)
+        is_single_measure = len(obs_val_cols) == 1
+
+        # If the cube is in pivoted shape, check what the column represents to set the aboutUrl
+        if self.is_cube_in_pivoted_shape:
+            # If the column represents a QbObservationValue, then simply assign the obs_val_column to this column.
+            if isinstance(column.structural_definition, QbObservationValue):
+                obs_val_col = column
+            # If the column represents an attribute, set the valueUrl using the _get_observation_value_col_for_column function
+            elif isinstance(column.structural_definition, QbAttribute):
+                if is_single_measure:
+                    obs_val_col = obs_val_cols[0]
+                else:
+                    col_title = (
+                        column.structural_definition.get_observed_value_col_title()
+                    )
+                    if col_title is not None:
+                        obs_val_col = self._get_observation_value_col_for_column(
+                            col_title
+                        )
+            # If the column represents units, set the valueUrl using the _get_observation_value_col_for_column function
+            elif isinstance(column.structural_definition, QbMultiUnits):
+                if is_single_measure:
+                    obs_val_col = obs_val_cols[0]
+                else:
+                    col_title = column.structural_definition.observed_value_col_title
+                    if col_title is not None:
+                        obs_val_col = self._get_observation_value_col_for_column(
```

Comment:
Since we already have `obs_val_cols`, can we pass it in to `_get_observation_value_col_for_column` so we don't need to duplicate effort?